handler/auth.go

```diff
@@ -206,16 +196,7 @@ func (h *Handler) VerifyKey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := h.db.Transaction(func(tx *gorm.DB) error {
-		registrationKey.User.Verified = true
-		if err := tx.Save(&registrationKey.User).Error; err != nil {
-			return err
-		}
-		if err := tx.Delete(&registrationKey).Error; err != nil {
-			return err
-		}
-		return nil
-	}); err != nil {
+	if err := h.usersRepo.Activate(registrationKey); err != nil {
```

Comment:
I think you misunderstood. I was saying to call `Activate` with the GUID itself, not the model.

Reply:
I most definitely did misunderstand